module/VuFind/src/VuFind/Db/Migration/MigrationLoader.php

```diff
@@ -85,16 +87,25 @@ public function getMigrationSubdirectoriesMatchingVersion(string $oldVersion, st
         }
 
         // Find version-appropriate subdirectories in the migration directory:
-        return array_values(
-            array_filter(
-                glob("$basePath/*"),
-                function ($path) use ($oldVersion) {
-                    $parts = explode('/', $path);
-                    $version = array_pop($parts);
-                    return preg_match('/^\d/', $version) && Comparator::greaterThanOrEqualTo($version, $oldVersion);
-                }
-            )
+        $directories = array_filter(
+            glob("$basePath/*"),
+            function ($path) use ($oldVersion) {
+                $parts = explode(DIRECTORY_SEPARATOR, $path);
+                $version = array_pop($parts);
+                return preg_match('/^\d/', $version) && Comparator::greaterThanOrEqualTo($version, $oldVersion);
+            }
+        );
+
+        usort(
+            $directories,
+            function ($a, $b): int {
+                $aParts = explode(DIRECTORY_SEPARATOR, $a);
+                $bParts = explode(DIRECTORY_SEPARATOR, $b);
+                return Comparator::greaterThan(end($aParts), end($bParts));
```

Comment:
This is currently casting a boolean to an integer, likely with unexpected results. Shouldn't it be more like this?


```suggestion
                $aParts = explode(DIRECTORY_SEPARATOR, $a);
                $bParts = explode(DIRECTORY_SEPARATOR, $b);
                $aPartEnd = end($aParts);
                $bPartEnd = end($bParts)
                return $aPartEnd === $bPartEnd ? 0 : Comparator::greaterThan($aPartEnd, $bPartEnd) ? 1 : -1;
```

Reply:
True. It worked somehow, but it's much better to be explicit.